uitools/import/Esri/ArcGISRuntime/Toolkit/Controller/BasemapGalleryController.qml

```diff
@@ -221,6 +248,21 @@ QtObject {
             }
         }
 
+        function setCurrentBasemapMatchingSp(basemap, layer) {
+            if(layer.loadStatus === Enums.LoadStatusLoaded) {
+                if(!basemapMatchesCurrentSpatialReference(basemap)){
+                    //manually setting the currbasemap to trigger the event onCurrentBasemapChaged
+                    internal.currentBasemap = internal.currentBasemap;
+                    geoModel.basemap = geoModel.basemap;
+                    internal.gallery.dataChanged(internal.gallery.index(basemapIndex(basemap), 0), internal.gallery.index(basemapIndex(basemap), 0));
+                    return;
+                }
+                internal.currentBasemap = basemap;
+                geoModel.basemap = internal.currentBasemap;
```

Comment:
❓ If the geoModel basemap is being set, does the `internal.currentBasemap` need set? I'm hoping it would update automatically.

Reply:
❓ Not sure what you mean exactly. If we don't set the `internal.currentBasemap`, then the `geoModel.basemap` will be set to the old `internal.currentBasemap`. Did you mean the other way around, to not set `geoModel.basemap` manually?

Reply:
There's a connections object at line 243 which I'd hope would keep the two in sync, as the geomodel basemap can be changed from outwith the gallery anyway.

Reply:
❗ found out that self assignment will not trigger the binding, so 
`geoModel.basemap = geoModel.basemap` will not set the `internal.currentBasemap`. This is shown by the busy indicator keep spinning for the New Zealand basemap, which is stopped upon the raise of onCurrentBasemapChanged()